app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HallsController < ApplicationController
+
+    before_action :set_hall, only: [:show, :update, :destroy]
+
+    def index
+        @halls = Hall.all.map do |hall|
+            {
+                name: hall.hall_number,
```

Comment:
I see at least three different naming - `name`, `hall_name` and `hall_number` - how about unifying this and using just a single name?